File: sdks/php/src/Resources/BranchesResource.php
<?php

namespace Puxbay\Resources;

use Puxbay\Models\Branch;
use Puxbay\Models\PaginatedResponse;

class BranchesResource extends BaseResource
{
    public function list(int $page = 1): PaginatedResponse
    {
        $data = $this->get("branches/?page={$page}");
        return PaginatedResponse::fromArray($data, fn($item) => Branch::fromArray($item));
    }

    public function getBranch(string $branchId): Branch
    {
        $data = $this->get("branches/{$branchId}/");
        return Branch::fromArray($data);
    }

    public function create(Branch $branch): Branch
    {
        $data = $this->post('branches/', $branch->toArray());
        return Branch::fromArray($data);
    }

    public function update(string $branchId, Branch $branch): Branch
    {
        $data = $this->patch("branches/{$branchId}/", $branch->toArray());
        return Branch::fromArray($data);
    }

    public function deleteBranch(string $branchId): void
    {
        $this->delete("branches/{$branchId}/");
    }
}

File: sdks/dotnet/src/Resources/BranchesResource.cs
using System.Threading.Tasks;
using Puxbay.SDK.Models;

namespace Puxbay.SDK.Resources
{
    public class BranchesResource
    {
        private readonly PuxbayClient _client;
        public BranchesResource(PuxbayClient client) { _client = client; }

        public async Task<PaginatedResponse<Branch>> ListAsync(int page = 1)
        {
            return await _client.GetAsync<PaginatedResponse<Branch>>($"branches/?page={page}");
        }

        public async Task<Branch> GetAsync(string branchId)
        {
            return await _client.GetAsync<Branch>($"branches/{branchId}/");
        }

        public async Task<Branch> CreateAsync(Branch branch)
        {
            return await _client.PostAsync<Branch>("branches/", branch);
        }

        public async Task<Branch> UpdateAsync(string branchId, Branch branch)
        {
            return await _client.PatchAsync<Branch>($"branches/{branchId}/", branch);
        }

        public async Task DeleteAsync(string branchId)
        {
            await _client.DeleteAsync($"branches/{branchId}/");
        }
    }
}
